src/librustc_mir/transform/cleanup_post_borrowck.rs

```diff
@@ -42,15 +42,15 @@ impl MirPass for CleanEndRegions {
                           tcx: TyCtxt<'a, 'tcx, 'tcx>,
                           _source: MirSource,
                           mir: &mut Mir<'tcx>) {
-        if !tcx.emit_end_regions() { return; }
+        if tcx.emit_end_regions() {
```

Comment:
any particular reason for this hunk, or just cleanup? (I sort of prefer the "return early" variant)

Reply:
So do I actually, this was just left-over from when I had `CleanEndRegions` and `CleanUserAssertTy` combined as a single pass and I didn't want an early return to stop the deletion of the `UserAssertTy`. I can change it back.